src/solvers/refinement/string_refinement.cpp

```diff
@@ -757,6 +757,7 @@ decision_proceduret::resultt string_refinementt::dec_solve()
 
   // Initial try without index set
   const auto get = [this](const exprt &expr) { return this->get(expr); };
+  dependences.clean_cache();
```

Comment:
I believe the plural is "dependencies". Notably, your code sometimes already uses "dependencies".

Reply:
my understanding is that dependences is the the plural of dependence and dependencies the plural of dependency. Anyway I'd prefer to keep only one...